packages/http-server-javascript/src/common/reference.ts

```diff
@@ -0,0 +1,296 @@
+      // might not be desirable.
+      const parentModule = createOrGetModuleForNamespace(ctx, effectiveModel.namespace);
+
+      module.imports.push({
+        binder: [templatedName.pascalCase],
+        from: parentModule,
+      });
+
+      return templatedName.pascalCase;
+    }
+    case "Union": {
+      if (type.variants.size === 0) return "never";
+      else if (type.variants.size === 1)
+        return emitTypeReference(ctx, [...type.variants.values()][0], position, module, options);
+
+      if (options.requireDeclaration) {
+        if (type.name) {
+          const nameCase = parseCase(type.name);
+
+          ctx.typeQueue.add(type);
+
+          module.imports.push({
+            binder: [nameCase.pascalCase],
+            from: createOrGetModuleForNamespace(ctx, type.namespace!),
+          });
+
+          return type.name;
+        } else {
+          const existingSyntheticName = ctx.syntheticNames.get(type);
+
+          if (existingSyntheticName) {
+            module.imports.push({
+              binder: [existingSyntheticName],
+              from: ctx.syntheticModule,
+            });
+
+            return existingSyntheticName;
+          } else {
+            const altName = options.altName;
+
+            if (!altName) {
+              throw new Error("UNREACHABLE: anonymous union without preferredAlternativeName");
+            }
+
+            ctx.synthetics.push({
+              kind: "anonymous",
+              name: altName,
+              underlying: type,
+            });
+
+            module.imports.push({
+              binder: [altName],
+              from: ctx.syntheticModule,
+            });
+
+            ctx.syntheticNames.set(type, altName);
+
+            return altName;
+          }
+        }
+      } else {
+        return emitUnionType(ctx, [...type.variants.values()], module);
+      }
+    }
+    case "Enum": {
+      ctx.typeQueue.add(type);
+
+      const name = parseCase(type.name).pascalCase;
+
+      module.imports.push({
+        binder: [name],
+        from: createOrGetModuleForNamespace(ctx, type.namespace!),
+      });
+
+      return name;
+    }
+    case "String":
+      return JSON.stringify(type.value);
+    case "Number":
+    case "Boolean":
+      return String(type.value);
+    case "Intrinsic":
+      switch (type.name) {
+        case "never":
+          return "never";
+        case "null":
+          return "null";
+        case "void":
+          // It's a bit strange to have a void property, but it's possible, and TypeScript allows it. Void is simply
+          // only assignable from undefined or void itself.
+          return "void";
+        case "ErrorType":
+          throw new Error("UNREACHABLE: Encountered ErrorType in emitTypeReference");
+        case "unknown":
+          return "unknown";
+        default:
+          throw new Error(
```

Comment:
any way this can be made a warning and return `unknown` instead of throwing an error

Reply:
Yes, I talked to Mark about this and seem to remember thinking it was a good idea to just treat anything we don't know as unknown. I can do the same treatment for scalars.